test/Libraries/RevitNodesTests/Elements/GroupTests.cs

```diff
@@ -0,0 +1,164 @@
+
+            // Act
+            var ungroupedElements = group.UngroupElements();
+            var ungroupedElementIds = ungroupedElements.Select(x => x.Id);
+
+            // Assert
+            CollectionAssert.AreEqual(expectedUngroupedElements, ungroupedElementIds);
+        }
+
+        [Test]
+        [TestModel(@".\Group\GroupTest.rvt")]
+        public void CanGetAttachedDetailGroup()
+        {
+            // Arrange
+            var modelGroup = ElementSelector.ByElementId(316699, true) as Group;
+            var attachedGroup = ElementSelector.ByElementId(316701, true) as Group;
+
+            var expectedAttachedDetailGroup = new List<int>() { 316702 };
+
+            // Act
+            var attachedDetailGroup = modelGroup.AttachedDetailGroup;
+            var attachedDetailGroupId = attachedDetailGroup.Select(x => x.Id);
+
+            // Assert
+            Assert.AreEqual(expectedAttachedDetailGroup, attachedDetailGroupId);
+        }
+
+        [Test]
+        [TestModel(@".\Group\GroupTest.rvt")]
+        public void CanGetGroupType()
+        {
+            // Arrange
+            var modelGroup = ElementSelector.ByElementId(316699, true) as Group;
+            var attachedGroup = ElementSelector.ByElementId(316701, true) as Group;
+
+            var expectedModelGroupTypeId = 316700;
+            var expectedattachedGroupTypeId = 316702;
+
+            // Act
+            var modelGroupTypeId = modelGroup.GroupType.Id;
+            var attachedGroupTypeId = attachedGroup.GroupType.Id;
+
+            // Assert
+            Assert.AreEqual(expectedModelGroupTypeId, modelGroupTypeId);
+            Assert.AreEqual(expectedattachedGroupTypeId, attachedGroupTypeId);
+
+
+        }
+
+        [Test]
+        [TestModel(@".\Group\GroupTest.rvt")]
+        public void CanCheckIfGroupIsAttached()
+        {
+            // Arrange
+            var modelGroup = ElementSelector.ByElementId(316699, true) as Group;
+            var attachedGroup = ElementSelector.ByElementId(316701, true) as Group;
+
+            // Assert
+            Assert.AreEqual(false, modelGroup.IsAttached);
+            Assert.AreEqual(true, attachedGroup.IsAttached);
+        }
+
+        [Test]
+        [TestModel(@".\Group\GroupTest.rvt")]
+        public void CanGetGroupLocation()
+        {
+            // Arrange
+            var modelGroup = ElementSelector.ByElementId(316699, true) as Group;
+            var attachedGroup = ElementSelector.ByElementId(316701, true) as Group;
+
+            var expectedModelGroupLocation = Point.ByCoordinates(-5838.369, 4122.718, 0.000);
+            var expectedAttachedGroupLocationExceptionMessage = Revit.Properties.Resources.AttachedGroupLocation;
+
+            // Act
+            var modelGroupLocation = modelGroup.Location;
+            var attachedGroupLocation = Assert.Throws<InvalidOperationException>(() => GetLocation(attachedGroup));
```

Comment:
you can probs replace `GetLocation(attachedGroup)` with `return attachedGroup.Location` and remove the private method if only used here